doc/introduction/interfaces.rst

```diff
@@ -3,17 +3,104 @@
+  that estimate measurement statistics using a finite number of shots; please use
+  the ``parameter-shift`` rule instead.
+
+* ``"adjoint"``: A form of backpropagation that takes advantage of the unitary or reversible
+  nature of quantum computation.
+
+  The `adjoint method <https://arxiv.org/abs/2009.02823>`__  reverses through the circuit after a
+  forward pass by iteratively applying the inverse (adjoint) gate. This method is similar to
+  ``"backprop"``, but trades off increased runtime with significantly lower memory usage.
+
+* ``"reversible"``: A form of backpropagation that takes advantage of the unitary or reversible
+  nature of quantum computation.
+
+  This method is similar to the ``adjoint`` method, but has a slightly larger time overhead and a similar
+  memory overhead.
+
+  Compared to the parameter-shift rule, the reversible method can be faster or slower, depending on
+  the density and location of parametrized gates in a circuit.
+
+Hardware-compatible differentiation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following methods support both quantum hardware and simulators, and are examples of `forward
+accumulation <https://en.wikipedia.org/wiki/Automatic_differentiation#Forward_accumulation>`__
+However, when using a simulator, you may notice that the time required to compute the gradients
+:doc:`scales quadratically <demos/tutorial_backprop>` with the number of trainable circuit
+parameters.
+
+* ``"parameter-shift"``: Use the analytic :doc:`parameter-shift rule
+  <glossary/parameter_shift>` for all supported quantum operation arguments, with
+  finite-difference as a fallback.
+
+* ``"finite-diff"``: Uses numerical finite-differences for all quantum operation arguments.
+
+
+Device gradients
+~~~~~~~~~~~~~~~~
+
+* ``"device"``: Queries the device directly for the gradient.
+  Only allowed on devices that provide their own gradient computation.
+
+
+.. note::
+
+    If not specified, the default differentiation method is ``diff_method="best"``. PennyLane
+    will attempt to determine the *best* differentiation method given the device and interface.
+    Typically, PennyLane will prioritize device-provided gradients, backpropagation, parameter-shift
+    rule, and finally finite-differences, in that order.
+
+
+Training and interfaces
+-----------------------
 
 The bridge between the quantum and classical worlds is provided in PennyLane via *interfaces*.
-Currently, there are four built-in interfaces: NumPy, PyTorch, JAX, and TensorFlow.
-These interfaces make each of these libraries quantum-aware, allowing quantum circuits to be
-treated just like any other operation.
+Currently, there are four built-in interfaces: :doc:`NumPy <interfaces/numpy>`, :doc:`PyTorch
+<interfaces/torch>`, :doc:`JAX <interfaces/jax>`, and :doc:`TensorFlow <interfaces/tf>`. These
+interfaces make each of these libraries quantum-aware, allowing quantum circuits to be treated just
+like any other operation.
```

Comment:
(minor) Below, it says: `(NumPy arrays, Torch Tensors, or TensorFlow Tensors)`. Might make more sense to use lowercase for `Tensors`, since it refers to the type/word tensor, rather than the `Tensor` object, right? 🤔